File: backend/Services/Parsers/SubwaydleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class SubwaydleParser : BasicScoreResultParser
    {
        private ILogger<SubwaydleParser> _logger;

        public SubwaydleParser(ILogger<SubwaydleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "Subwaydle";
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($"{GameName} \\d+ \\(?[^)]*\\)? ?(?<{ScoreGroup}>[\\d|X])/\\d");
        protected override string? ExtraContent => null;
        public override string Url => "https://www.subwaydle.com/";
    }
}

File: lambdas/src/parsers/subwaydle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class SubwaydleParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Subwaydle';
  readonly helpText = null;
  protected readonly extraContent = null;
  readonly url = 'https://www.subwaydle.com/';
  
  // Matches "Subwaydle 727 X/6" or "Subwaydle 274 (Weekend Edition) 1/6"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `${this.gameName} \\d+ \\(?[^)]*\\)? ?(?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
